top left menu alerts - open mode - style updates

yinon-sh-work opened the pull request.

Commit:
modify TopLeftAlertMenu button to last figma v

includ:

- udpate primary default color in tailwind.config to the new color
- add muted alert icon
- disable default css button style in index.css file
- add text to he/translation.json file for the text button
- fix the load path in i18n.ts file

include:

- css and tailwind changes
- add states for dinamic vars

Commit:
modify TopMenu (Header) to last figma style

inclde:

- remvoe change language options
- change the bg color
- add header text to he/translation.json file

1shaked committed:
feat: add AddOperationalCommand component with form validation using react-hook-form

1shaked committed:
feat: enhance AddOperationalCommand component with improved styling and form structure; integrate into GnatAndPMButtonMenu

1shaked committed:
feat: enhance AddOperationalCommand component with date and time formatting; integrate moment.js for better time handling

1shaked committed:
feat: implement Technics form with validation and initial values; enhance AddOperationalCommand component structure

1shaked committed:
feat: enhance TechnicCW and TechnicSH components with improved styling and headings

Commit:
alert open style udpate and modify to last figma ver

include:

- TopleftAlertMenu component -  style updates
- ListOfAlerts component - gradient style to the alerts
- add translation to transslation file that needed
- modify the colors to last figma ver

1shaked committed:
feat: add ChoosePath component and Accordion component; enhance AddOperationalCommand structure

1shaked commented on src/Pages/Home/components/ListOfAlerts.tsx:
```diff
@@ -1,61 +1,72 @@
-                <div className=" bg-active text-active-foreground p-1 rounded">not read</div>
-                <div className="p-1 rounded">archive</div>
+        <div className="flex justify-between text-sm text-center">
+            <div className="flex gap-1 bg-secondary p-1 rounded-[20px]">
```
it is better for us to use relative units

Meirasulin replied:
Done.
See the PR in the internal network.

1shaked commented on src/Pages/Home/components/TopLeftAlertMenu.tsx:
```diff
@@ -5,35 +5,43 @@ import { RightArrowSvg } from "../../../components/Icons/RightArrow"
-        return <div className="bg-primary text-primary-foreground w-[25vw] relative">
-            <div className="flex justify-end items-center gap-2 ">
-                <div> Alerts and threats </div>
+        return <div className="bg-[#0d1d3599] shadow-[9px_0px_12.4px_0px_rgba(0,0,0,0.25)] backdrop-blur-[35px] rounded-[20px] text-primary-foreground w-[25vw] relative">
```
relative units only and prefer if needed custom shadow across the app then we will use custom class

1shaked replied:
also the color is hard coded

Meirasulin replied:
Done
See the PR in the internal network.

1shaked commented on src/Pages/Home/components/TopLeftAlertMenu.tsx:
```diff
@@ -5,35 +5,43 @@ import { RightArrowSvg } from "../../../components/Icons/RightArrow"
+    
     const [isAlertOpen, setIsAlertOpen] = useState(false)
+    const [isAlertMuted, setIsAlertMuted] = useState(false)
+    const [unreadAlerts] = useState(0)
```
this line is not needed cause we will have this from a query please delete if can

Meirasulin replied:
Done
See the PR in the internal network.

Commit:
fix relative styles and custom colors

- replace px to relative sizes
- add custom colors and shadows to tailwind.config.js file
- add custom text sizes (that configure in Figma) to tailwind.config.js file

:

Commit:
Merge branch 'dev_shaked' into dev_meir_alerts_style_30/10

Commit:
fix relative styles and custom colors in TopLeftAlertMenu

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+    return <div>
+        <select {...form.register('technicsUsed')} multiple onChange={(e) => {
+            // This is a rare case where we need to cast the selected values to the correct type because the type of the value is string but you know that the value is a TechnicsTypes
+            const selectedValues = Array.from(e.target.selectedOptions, option => option.value) as TechnicsTypes[];
```
this is to long function to put in the html 
please put it out and make a reference for the function and use it on the html

Meirasulin replied:
This comment should be in shake's PR.
I only did a pull to my branch, to update my brach to the main branch, that we agreed that for now it's shaked branch.

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+            console.log(selectedValues)
+            // This is the way to set the value of the form
+            if (technics === undefined) {
+                for (const v of selectedValues) {
```
u can add it to useEffect or different technic for it 
please make change
@1shaked  please take a look as well

Meirasulin replied:
Should be in shaked pr

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+    })),
+    
+})
+
```
1.why do you need export to each one? are we export each one of them?
if it is, you can make a group of them and the end of the file without write export each time (on different file or this file but put it separate)
2.all this constans should put it on different file and make inport to this file, please change it

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+
+export const ModulationSchemaZod = z.union([z.literal('AM'), z.literal('FM'), z.literal('PM')])
+export type ModulationType = z.infer<typeof ModulationSchemaZod>;
+export const ModulationArray: ModulationType[] = ['AM', 'FM', 'PM'] as const;
```
why did you add as  const at the end?
The 'as const' assertion already tells TypeScript to treat the array as a tuple of literal types, so you don't need to annotate it explicitly with ModulationType[]
so you can write it like this:
export const ModulationArray = ['AM', 'FM', 'PM'] as const;

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+    modulation: 'AM',
+    LO: 0
+} ;
+export const TechnicSHZod = z.object({
```
same here

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+
+})
+
+export type TechnicSHType = z.infer<typeof TechnicSHZod>;
```
same here in those lines

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+export type TechnicsFormType = z.infer<typeof TechnicsFormZod>;
+
+export type t = typeof TechnicsList
+interface AddOperationalCommandFormI {
```
this is not the right place for interface
we use a different file for interfaces

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+        </form>
+    </FormProvider>
+
+}
```
wrong place to put those imports 
please edit it to the up of the page

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+import moment from "moment";
+import { DATE_FORMAT, TIME_FORMAT } from "../../../utils/const";
+
+interface FormStepsProps {
```
same here

itusheve commented on src/Pages/Home/components/AddOperationalCommand.tsx:
```diff
@@ -0,0 +1,272 @@
+
+}
+
+export function CategoryDetails() {
```
after the return 
1.please do write any more function only before the return 
2. you can write this function in better way
you can use here useFormContext for better readable code

itusheve commented on src/Pages/Home/components/GnatAndPMButtonMenu.tsx:
```diff
@@ -2,10 +2,14 @@ import { useState } from "react"
     const [isGnatOpen, setIsGnatOpen] = useState(false)
+    const [isPMOpen, setIsPMOpen] = useState(false)
     return <div>
+      {isPMOpen && <AddOperationalCommand />}
```
this return its not write correctly:
we need to write it more cleaner and structured 
like this:

```
return (
    <div>
        {isPMOpen && (
            <>
            <AddOperationalCommand `/>`
                <span>new PM</span>
            </>
        )}
        <div className="flex justify-between">
            <button
                className="p-2 bg-primary text-primary-foreground flex gap-2"
                onClick={() => setIsGnatOpen(!isGnatOpen)}
            >
                <DoubleArrowDown
                    className={`transition-all ${isGnatOpen ? 'rotate-180' : ''}`}
                />
            </button>
           ......
        </div>
    </div>
);
```

itusheve commented on src/Pages/Home/components/ListOfAlerts.tsx:
```diff
@@ -1,61 +1,68 @@
 import { AlertIssueSvg } from "../../../components/Icons/AlertIssueSvg";
 import { WarnSvg } from "../../../components/Icons/WarnSvg";
 
+
```
delete this line please

Meirasulin replied:
From now we have eslint for issues like this.
in the Internal Network it's Done.

itusheve commented on src/Pages/Home/components/ListOfAlerts.tsx:
```diff
@@ -1,61 +1,68 @@
-                <div className="p-1 rounded">archive</div>
+        <div className="flex justify-between text-sm text-center">
+            <div className="flex gap-1 bg-secondary p-1 rounded-3xl">
+                <button className="bg-active text-active-foreground p-1 rounded-3xl transition-all duration-300 hover:scale-105">{t('TopLeftAlertMenu.ListOfAlerts.Unread')}</button>
```
use eslint to fix the line

itusheve commented on src/Pages/Home/components/ListOfAlerts.tsx:
```diff
@@ -1,61 +1,68 @@
-                        <div className="text-[#7C7C7C]">Wednesday 12:03</div>
+                <h2 className="text-center p-1">יום ד’ 18/12/2024</h2>
+                <div className="flex flex-col gap-2">
+                    <div className="border-r-[6px] rounded-3xl border-r-alert-border-issue bg-gradient-to-tr from-alert-gradient-issue-from from-[1.85%] to-alert-gradient-issue-to to-[97.51%] text-alert-foreground p-2">
```
same here

itusheve commented on src/Pages/Home/components/ListOfAlerts.tsx:
```diff
@@ -1,61 +1,68 @@
                                     <div>
                                         תקלה!
-                                        אמצעי SDA-A, RX:0 הגיע לטמפרטורה גבוהה, בוצע כיבוי אוטומטי. 
+                                        אמצעי SDA-A, RX:0 הגיע לטמפרטורה גבוהה, בוצע כיבוי אוטומטי.
```
please use en file to make translate and dont use hebrew on the code

Meirasulin replied:
This Text is only for example and its not, and for now until the front communicate with the back its hardcoded.

itusheve commented on src/Pages/Home/components/ListOfAlerts.tsx:
```diff
@@ -1,61 +1,68 @@
                                     <div>
                                         תקלה!
-                                        אמצעי SDA-A, RX:0 הגיע לטמפרטורה גבוהה, בוצע כיבוי אוטומטי. 
+                                        אמצעי SDA-A, RX:0 הגיע לטמפרטורה גבוהה, בוצע כיבוי אוטומטי.
```
same here

Meirasulin replied:
Same.

itusheve commented on src/Pages/Home/components/ListOfAlerts.tsx:
```diff
@@ -1,61 +1,68 @@
-                    <div className=" bg-alert text-warning-foreground border-warning-foreground border-2 py-1 px-2 rounded-lg">
-                        <div className="text-[#7C7C7C]">Wednesday 12:03</div>
+
+                    <div className="border-r-[6px] rounded-3xl border-r-alert-border-warn bg-gradient-to-tr from-alert-gradient-warn-from from-[1.85%] to-alert-gradient-warn-to to-[97.51%] text-alert-foreground p-2">
```
eslint here as well

itusheve commented on src/Pages/Home/components/TopLeftAlertMenu.tsx:
```diff
@@ -5,35 +5,38 @@ import { RightArrowSvg } from "../../../components/Icons/RightArrow"
     const [isAlertOpen, setIsAlertOpen] = useState(false)
+
     if (isAlertOpen) {
-        return <div className="bg-primary text-primary-foreground w-[25vw] relative">
```
here do you have duplicate  return and its wrong :
your code had two return statements within the same if block, which isn’t allowed. Consolidating this under a single if block keeps it clean.

please do more simple the code : wrapping the logic under the isAlertOpen condition and then using a separate return block for the default (when isAlertOpen is false) makes the structure more organized and readable

Meirasulin replied:
Please Inform shaked to don't write like this.
And in the Internal Network i already changed.

itusheve commented on src/Pages/Home/components/TopMenu.tsx:
```diff
@@ -6,30 +6,23 @@ import { useTranslation } from "react-i18next";
 export function TopMenu() {
-    const { i18n } = useTranslation();
-
+    const { t } = useTranslation();
```
why you change it?
please explain

itusheve commented on src/components/Accordion.tsx:
```diff
@@ -0,0 +1,3 @@
+export function Accordion() {
```
please remove i dont want to push empty function

itusheve commented on src/utils/const.ts:
```diff
@@ -112,4 +112,8 @@ export const MENU_LIST = [
     }
-] as const
\ No newline at end of file
+] as const
```
tipe 
maybe think to write it like that:
```
export const FORMATS = {
    DATE: 'DD/MM/YYYY',
    TIME: 'HH:mm',
} as const;
```

itusheve reviewed (changes requested):
please fix what i ask to change 
thanks
